src/App.js

```diff
@@ -86,10 +88,22 @@ const App = () => {
               </RequireAuth>
             }
           />
+          <Route
+            path="/profile/:userId"

```

Comment:
instead of wrapping every route to Require auth separately you can try and do it all one.
<RequireAuth>
 <Component1/>
<Component2/>
</RequireAuth>